Libraries/LibWeb/Layout/TreeBuilder.cpp

```diff
@@ -277,14 +283,39 @@ void TreeBuilder::create_pseudo_element_if_needed(DOM::Element& element, CSS::Ps
             || pseudo_element_content.type == CSS::ContentData::Type::None))
         return;
 
+    // For ::marker with content or display 'none' -- do nothing.
+    if (pseudo_element == CSS::PseudoElement::Marker
+        && (pseudo_element_display.is_none() || pseudo_element_content.type == CSS::ContentData::Type::None))
+        return;
+
+    // For ::marker with content 'normal', create the marker pseudo-element from a ListItemMarkerBox
+    // FIXME: This + ListItemBox + ListItemMarkerBox will disappear once ::marker pseudo-elements with 'normal' content
+    //        are rendered using the special list-item counter.
+    //        See: https://github.com/LadybirdBrowser/ladybird/issues/4782
+    if (element.layout_node() && is<ListItemBox>(*element.layout_node()) && pseudo_element == CSS::PseudoElement::Marker && pseudo_element_content.type == CSS::ContentData::Type::Normal) {
+        auto layout_node = element.layout_node();
```

Comment:
it seems you could also benefit from using `as_if` on `element.layout_node()` directly to get relevant `layout_node` used inside (esp. that you later also cast this into `ListItemBox`).